### PR TITLE
ComboBox: Use Popover v2

### PR DESCRIPTION
**Pull request by diyorbek**
### Summary

Removed all the Popover v2 experiment related code from ComboBox.

### Links

- [Jira](https://jira.pinadmin.com/browse/GWEB-65)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers, relevant feature teams)

### CODE REVIEW

File: packages/gestalt/src/ComboBox.js [hunk 1]
```diff
@@ -503,30 +497,28 @@ const ComboBoxWithForwardRef: AbstractComponent<Props, HTMLInputElement> = forwa
...
             onKeyDown={handleKeyDown}
-            positionRelativeToAnchor={false}
+            role="listbox"
             shouldFocus={false}
             size="flexible"
           >
             <Box
               ref={dropdownRef}
               alignItems="center"
-              aria-expanded={showOptionsList}
```
Comment (diyorbek) [inline packages/gestalt/src/ComboBox.js:521]: aria-expanded belongs to the controller of an expandable component

File: packages/gestalt/src/ComboBox.jsdom.test.js [hunk 2]
```diff
@@ -124,38 +121,37 @@ describe('ComboBox', () => {
...
 
-    it('renders disabled state', () => {
+    it('renders disabled state', async () => {
       const { baseElement } = renderComboBox({ disabled: true });
       expect(baseElement).toMatchSnapshot();
 
       expect(screen.queryByRole('button')).not.toBeInTheDocument();
 
-      fireEvent.click(screen.getByLabelText(LABEL));
+      await userEvent.click(screen.getByLabelText(LABEL));
```
Comment (diyorbek) [inline packages/gestalt/src/ComboBox.jsdom.test.js:130]: we should always try to use userEvent over fireEvent whenever we are able to

File: packages/gestalt/src/__snapshots__/ComboBox.jsdom.test.js.snap [hunk 3]
```diff
@@ -135,7 +135,7 @@ exports[`ComboBox Controlled ComboBox with Tags renders with tags 1`] = `
                     class="box"
                   >
                     <button
-                      class="closeButton default hideOutline tapTransition accessibilityOutline sm"
```
Comment (diyorbek) [inline packages/gestalt/src/__snapshots__/ComboBox.jsdom.test.js.snap:138]: Tag remove buttons cannot have outline all at the same time.

**Review by AlbertCarreras — APPROVED** [review]

File: packages/gestalt/src/ComboBox.jsdom.test.js (hunk 2 above)
Comment (rlingineni) [reply packages/gestalt/src/ComboBox.jsdom.test.js:130]: perfect

**Pull request merged**